worker/src/main/java/gov/cms/ab2d/worker/config/RoundRobinBlockingQueue.java

```diff
@@ -0,0 +1,357 @@
+package gov.cms.ab2d.worker.config;
+
+import lombok.extern.slf4j.Slf4j;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.*;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.Predicate;
+
+@Slf4j
+public class RoundRobinBlockingQueue<E> implements BlockingQueue<E> {
+    // The list of categories
+    private List<String> categories = new ArrayList<>();
+    // The individual category queues
+    private Map<String, LinkedList<E>> categoryQueues = new HashMap<>();
```

Comment:
You don't really need to maintain two separate Collections -- categories and queues -- and keep them in sync throughout the code. All you need is a "list of queues", i.e. `List<List>` or `List<Queue>` or something similar. This will definitely help with making this class more generic and unaware of its "surrounding context".